src/wms/src/main/java/org/geoserver/wms/icons/IconProperties.java

```diff
@@ -166,8 +173,15 @@ public String href(String baseURL, String workspace, String styleName) {
                                                 .substring(styles.getAbsolutePath().length() + 1);
                                 file = new File(relativePath);
                             } else {
-                                // we wont' transform this, other dirs are not published
-                                file = null;
+                                // other dirs are not published, serialize uri-data
+                                Path path = file.toPath();
+                                String contentType = Files.probeContentType(path);
+                                byte[] bytes = Files.readAllBytes(path);
+                                return new StringBuilder("data:")
+                                        .append(contentType)
+                                        .append(";base64,")
+                                        .append(Base64.getEncoder().encodeToString(bytes))
+                                        .toString();
```

Comment:
Relative references should work too though, the contents of the styles directory is available to the users.
So for example, the "grass_poly.sld" [file in the release data directory](https://github.com/geoserver/geoserver/blob/main/data/release/styles/grass_poly.sld) contains this relative reference:

                    <OnlineResource xlink:type="simple" xlink:href="grass_fill.png" />

which is accessible using a link to the origin GeoServer, e.g.:

https://gs-main.geosolutionsgroup.com/geoserver/styles/grass_fill.png

Long story short, it seems to me the "file.isAbsolute" test is too strict